### PR TITLE
installer: skip old revisions and report stucked installer pods via events

### PR DESCRIPTION
**Pull request by mfojtik**
In case there is newer revision available, we should skip creating the installer pod for revision that is  already "old". Instead, we can wait for revision controller to update node target revision to match the latest level...

This might fix the case when we creating revision 5 pod while revision 7 is already available.

Additionally this add commit that will report Warning event when the installer pod is stucked in Pending for longer than 10 minutes. The event will carry reason and message from the pod.

### CODE REVIEW

File: pkg/operator/staticpod/controller/installer/installer_controller.go [hunk 1]
```diff
@@ -300,6 +300,12 @@ func (c *InstallerController) manageInstallationPods(operatorSpec *operatorv1.St
 
 		// if we are in a transition, check to see whether our installer pod completed
 		if currNodeState.TargetRevision > currNodeState.CurrentRevision {
+			// Skip creating the installer pod when there is already a revision higher than target revision
+			// In this case, wait for the revision controller to update the target revision on node.
+			if operatorStatus.LatestAvailableRevision > currNodeState.TargetRevision {
```
Comment (sttts) [inline pkg/operator/staticpod/controller/installer/installer_controller.go]: expecting a test to reflect this? I thought this has good test coverage.

Comment (mfojtik) [reply pkg/operator/staticpod/controller/installer/installer_controller.go]: this function has zero test coverage... I added tests in last commit.

**Comment by mfojtik** [comment]
/hold

This is pending proof-bump in one of the static pod operators but it is ready for review.

**Comment by mfojtik** [comment]
per @deads2k suggestion, I will rewrite this to separate controller that will observe the status of installer pods and report staled/pending conditions up to the operator condition levels.

**Pull request closed without merging**